modules/minimongo-observable/observable-cursor.ts

```diff
@@ -79,13 +80,31 @@ export class ObservableCursor<T> extends Observable<T[]> {
     });
   }
 
+  addedAt(doc, at, before) {
+    this.data.splice(at, 0, doc);
+    this.handleChange();
+  }
+
+  changedAt(doc, old, at) {
+    this.data[at] = doc;
+    this.handleChange();
+  };
+
+  removedAt(doc, at) {
+    this.data.splice(at, 1);
+    this.handleChange();
+  };
+
+  handleChange() {
+    this._runNext(this._cursor);
+  };
+
   _observeCursor(cursor: Mongo.Cursor<T>) {
-    const handleChange = () => { this._runNext(cursor); };
     return gZone.run(
-      () => cursor.observeChanges({
-        added: handleChange,
-        changed: handleChange,
-        removed: handleChange
+      () => cursor.observe({
+        addedAt: this.addedAt.bind(this),
```

Comment:
Can you pls put here handleChange for each method  as it was before, and check if it works for you?